src/detectors/builtin/sendInLoop.ts

```diff
@@ -14,42 +14,36 @@ import {
   AstStatementUntil,
   AstStatementForEach,
   AstFunctionDef,
-  AstAsmFunctionDef,
+  AstReceiver,
+  AstContractInit,
+  idText,
 } from "@tact-lang/compiler/dist/grammar/ast";
 
 /**
- * A detector that identifies send function calls inside loops and loops that may cause out-of-gas exceptions.
+ * A detector that identifies send function calls inside loops.
  *
  * ## Why is it bad?
  * **Send Functions Inside Loops:**
- * - **Unexpected Behavior:** Multiple sends may occur, leading to unintended contract interactions.
- * - **High Gas Consumption:** Loops with send calls can consume excessive gas, potentially causing out-of-gas exceptions.
- * - **Reentrancy Vulnerabilities:** Sending messages within loops might open up reentrancy attack vectors.
+ * - **Unexpected Behavior:** Multiple sends within a loop may result in sending messages multiple times, which could be unintended. For example, sending funds repeatedly might deplete contract balances or cause logic errors.
+ * - **High Gas Consumption:** Loops with send calls can consume excessive gas.
  *
- * **Loops with Excessive Iterations:**
- * - **Out-of-Gas Exceptions:** Loops with a large number of iterations may consume all available gas.
- * - **Denial of Service:** Excessive gas consumption can be exploited to create DoS attacks.
+ * **Recommendation:** Refactor the code to avoid calling send functions inside loops.
  *
  * ## Example
  * ```tact
  * // Send function inside a loop
  * while (i < limit) {
  *   send(SendParameters{ to: recipient, ... });
  * }
- *
- * // Loop with excessive iterations
- * repeat (1_000_001) {
```

Comment:
By the way, it might be an interesting idea for a new optional detector. It is very easy to implement, but might be useful in some cases.

Reply:
Yes on future we need optional detector.

Reply:
Could you create an issue then?